src/broker/cas_execute.c

```diff
@@ -8380,12 +8380,61 @@ sch_class_info (T_NET_BUF * net_buf, char *class_name, char pattern_flag, char v
       STRING_APPEND (sql_p, avail_size, "AND class_type = 'VCLASS' ");
     }
 
+  if (schema_name[0] == '\0')
+    {
+      strncpy (schema_name, database_user, DB_MAX_SCHEMA_LENGTH);
+    }
+
   if (pattern_flag & CCI_CLASS_NAME_PATTERN_MATCH)
     {
       if (class_name_only)
 	{
-	  STRING_APPEND (sql_p, avail_size, "AND class_name LIKE '%s' ESCAPE '%s' ", class_name_only,
-			 get_backslash_escape_string ());
+	  STRING_APPEND (sql_p, avail_size, "AND ");
+          // *INDENT-OFF*
+          STRING_APPEND (sql_p, avail_size,
+	    "(CASE WHEN "
+	        "(SELECT "
+	            "count (*) "
+	        "FROM "
+	            "db_synonym "
+	        "WHERE "
+	            "synonym_name LIKE '%s' ESCAPE '%s' "
+                    "AND synonym_owner_name = UPPER ('%s')) > 0 "
```

Comment:
As for synonyms, only private synonyms exist.
Therefore, in db_synonym, only synonyms owned by the user of the current session can be retrieved.
No predicate required for synonym_owner_name.